### PR TITLE
[DEV-1553] [DEV-1501] [DEV-1566] Deploy auction contracts for diamond

### PR DESCRIPTION
**Pull request by viet10101988**
- Contract v4 for Polygon and BSC

### CODE REVIEW

File: src/config/sdk.ts [hunk 1]
```diff
@@ -26,10 +26,12 @@ export const getContractsTags = (
       ];
     case Environment.Testnet:
       return [
-        chain === ChainType.BSC
+        chain === ChainType.POLYGON
```
Comment (Superjo149) [inline src/config/sdk.ts:29]: Its already deployed for BSC, so why not leave it also? We'll need to align everything in the end either way.

File: src/nft/AbstractNFT.ts [hunk 2]
```diff
@@ -196,14 +195,34 @@ export abstract class AbstractNFT {
...
+    const isDiamondContract = currentAuctionContract.hasTagSemver(
+      "AUCTION",
+      ">=4.0.0"
+    );
+
+    // We are using tranferProxy for diamondContracts so need to approve the address
+    if (isDiamondContract) {
+      const addressForApproval = this.transferProxyContract.address;
+      await this.approveIfNeeded(addressForApproval);
+    }
```
Comment (Superjo149) [inline src/nft/AbstractNFT.ts:212]: The line above `await this.approveIfNeeded(this.auctionContract.address);` should be included in the else statement here. No need to approve the auction contracts anymore in this case.

Or to make it cleaner
```ts
const addressToApprove = isDiamondContract ? this.transferProxyContract.address : this.auctionContract.address;
await this.approveIfNeeded(addressToApprove);
```

File: src/nft/AbstractNFT.ts [hunk 3]
```diff
@@ -255,57 +274,56 @@ export abstract class AbstractNFT {
...
+        if (!auctionId) {
+          auctionId = await this.getAuctionId(auctionContractAddress);
+        }
 
-      result = await ethersContracts.placeBid(auctionId, ...valueParam);
+        assert(!!auctionId, "AuctionId must be defined");
+        result = await ethersContracts.placeBid(auctionId, ...valueParam);
+      }
+      return result;
+    } catch (err) {
```
Comment (Superjo149) [inline src/nft/AbstractNFT.ts:324]: Please remove the try catch

**Review by Superjo149 — CHANGES_REQUESTED** [review]

File: src/config/sdk.ts (hunk 1 above)
Comment (deveshrawat) [reply src/config/sdk.ts:29]: Only sale contracts are deployed, so can keep that, but I would prefer testing it on polygon first with sale/auction then deploying contracts on BSC ?

Comment (Superjo149) [reply src/config/sdk.ts]: Its easier the sooner we do it, because if we do it before we go to testnet. You can just include it in the other migration

**Review by Superjo149 — APPROVED** [review]

**Pull request merged**